openwisp_radius/tests/test_api.py

```diff
@@ -890,6 +894,65 @@ def test_api_batch_add_users(self):
             with self.subTest(test_user=test_user):
                 self.assertIn((self.default_org.pk,), test_user.organizations_pk)
 
+    def test_api_batch_pdf_link(self):
+        post_url = f'{reverse("radius:batch")}{self.token_querystring}'
+        response = self.client.post(
+            post_url,
+            {
+                'name': 'test-prefix18',
+                'prefix': 'test-prefix18',
+                'number_of_users': 5,
+                'strategy': 'prefix',
+            },
+        )
+        pdf_link = response.json()['pdf_link']
+        with self.subTest('No Login'):
+            pdf_response = self.client.get(pdf_link)
+            self.assertEqual(pdf_response.status_code, 401)
+        with self.subTest('Login: normal user'):
+            self.client.force_login(self._get_user())
+            pdf_response = self.client.get(pdf_link)
+            self.assertEqual(pdf_response.status_code, 403)
+        with self.subTest('Login: operator without permission'):
+            operator = self._get_operator()
+            self.client.force_login(operator)
+            pdf_response = self.client.get(pdf_link)
+            self.assertEqual(pdf_response.status_code, 403)
+        with self.subTest('Login: operator without organization'):
+            view_radbatch_perm = Permission.objects.get(codename='view_radiusbatch')
+            operator.user_permissions.add(view_radbatch_perm)
+            pdf_response = self.client.get(pdf_link)
+            self.assertEqual(pdf_response.status_code, 400)
+        with self.subTest('Login: operator allowed'):
+            self._create_org_user(
+                **{'organization': self.default_org, 'user': operator}
+            )
+            pdf_response = self.client.get(pdf_link)
+            self.assertEqual(pdf_response.status_code, 200)
+
+    def test_batch_csv_pdf_link_404(self):
+        self.assertEqual(RadiusBatch.objects.count(), 0)
+        self.assertEqual(User.objects.count(), 0)
+        path_csv = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), 'static', 'test_batch.csv'
+        )
+        with open(path_csv, 'rt') as file:
+            data = self._get_post_defaults(
+                {'name': 'test_csv1', 'strategy': 'csv', 'csvfile': file}
+            )
+            response = self.client.post(
+                reverse('radius:batch'), data, HTTP_AUTHORIZATION=self.auth_header
+            )
+        self.assertEqual(response.status_code, 201)
+        response_json = json.loads(response.content)
+        pdf_url = reverse(
+            'radius:download_rad_batch_pdf',
+            args=[response_json['organization'], response_json['id']],
+        )
+        self.assertEqual(response_json['pdf_link'], None)
+        pdf_response = self.client.get(pdf_url)
+        self.assertEqual(pdf_response.status_code, 404)
```

Comment:
you're testing only that the strategy is prefix but not checking a non existing valid UUID, nor an invalid UUID

note: the invalid UUID check may be avoided if we use path: https://docs.djangoproject.com/en/3.0/ref/urls/#path
As we're doing here:  https://github.com/openwisp/openwisp-firmware-upgrader/pull/84